matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/extensions/RealmExtensions.kt

```diff
@@ -30,7 +30,11 @@ internal fun RealmObject.assertIsManaged() {
  */
 internal fun <T> RealmList<T>.clearWith(delete: (T) -> Unit) {
     while (!isEmpty()) {
+        val previousSize = size
         first()?.let { delete.invoke(it) }
+        if (previousSize != size + 1) {
```

Comment:
should this be `previousSize == size`? (to highlight no deletion has occurred)

Reply:
I see your point, but I expect `previousSize == size + 1` to be the only valid condition, so I think it's better like that.

We may have lambda which add object(s) to the list, or delete multiple objects, we live in a crazy world!

This will be possible to delete another item than the provide one in the lambda, for instance the first one, but at the end this will have the same result.

Another solution to avoid that would have to change the Extension to:

```kotlin
internal fun <T : RealmObject /* <- Needed contract to be able to use `isValid` */> RealmList<T>.clearWith(delete: (T) -> Unit) {
    while (!isEmpty()) {
        first()?.let {
            delete.invoke(it)
            if (it.isValid) {
                // Object has not been deleted
                error("`clearWith` MUST delete all elements of the RealmList")
            }
        }
    }
}
```

or even just delete it with a warning:

```kotlin
internal fun <T : RealmObject /* <- Needed contract to be able to use `isValid` */> RealmList<T>.clearWith(delete: (T) -> Unit) {
    while (!isEmpty()) {
        first()?.let {
            delete.invoke(it)
            if (it.isValid) {
                // Object has not been deleted
                Timber.w("`clearWith` MUST delete all elements of the RealmList")
                it.deleteFromRealm()
            }
        }
    }
}
```

Reply:
I do not think we can use `map`, because the RealmList is getting empty while we call `deleteFromRealm()` on its items. This is not really Kotlin logic.
I will add some Kdoc on the fun.

Reply:
Oh i understand your code with `map` better now, have read too fast. It may work. Let me give it a try.

Reply:
@ouchadam @mnaturel , I have updated the PR, and tested again dummy migration, it behaves at expected.